docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -565,6 +584,20 @@ plugin libraries, composed of:
   - It also implements OCALLs for the enclave-side library to access
   host-side services for SGX quote verification.
 
+The OE SDK V0.10 release will also have enclave-side and host-side plugins
+for verification of SGX ECDSA reports generated by legacy API function
+`oe_get_report()` and raw SGX ECDSA quotes generated by the Intel SGX SDK.
+These two evidence formats are identified by their UUIDs:
+- `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT`: for SGX ECDSA report generated
+by `oe_get_report()`.
+- `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE`: for raw SGX ECDSA quote generated
+by the Intel SGX SDK.
+
+For evidence data generated by `oe_get_evidence()`, the list of claims returned
+by `oe_verify_evidence()` will contain one claim named
```

Comment:
```suggestion
by `oe_verify_evidence()` can contain one claim named
```
It would only have such a claim if the set of custom claims was not the empty set, correct?

Reply:
What I assume is that `OE_CLAIM_CUSTOM_CLAIMS_ID` is always present in this case, but its value could be a zero-sized buffer.

In practice, either way would work, though the verifier application logic need to handle the error cases slightly differently.

Reply:
I would prefer a behavior (for ALL claims) that the claims are only returned if they're present in the evidence.   My second choice would be for the behavior to be the same for ALL claims, so that if you return claims with a 0-size buffer, then you do the same for ALL possible claims, but I like that behavior much less.   I would argue against having the behavior be different for different clams.

Reply:
I'm ok not to output an empty custom claims buffer. Let me update the document to reflect that.

The second choice would be hard to implement for those claims that are only applicable to a subset of formats. One example is `OE_CLAIM_SGX_REPORT_DATA_ID` claim.

Reply:
I'm ok with outputting an empty custom claim buffer if it was not specified when the evidence was generated.

Reply:
A zero-sized custom claims buffer is still valid, and is hashed in the `oe_get_evidence()` implementation. As a result, it still make more sense for `oe_verify_evidence()` to return a zero_sized custom claims entry. This is a statement by `oe_verify_evidence()` that there is a zero-sized custom claims buffer, instead of it being absent in the quote.

Reply:
What purpose would inserting a zero-sized claims buffer into the serialized evidence buffer, as opposed to omitting it?   Can we just declare it an invalid parameter to pass in a zero-sized buffer?

Reply:
`oe_get_evidence()` should allow its caller to not pass any custom claim (i.e. to pass an empty buffer). In this case, if `oe_verify_evidence()` does not return a custom claims entry with an empty buffer, could there be a vulnerability for an attacker to inject its favorite custom claims? I don't think there is vulnerability here, but like to be sure.